lib/nested_filter.ex

```diff
@@ -5,57 +5,45 @@ defmodule NestedFilter do
 
   @doc """
   Take a (nested) map and filter out any keys with specified values in the
-  filter_values list.
+  values_to_reject list.
   """
-  @spec drop_by_value(map :: map(), filter_values :: list()) :: map()
-  def drop_by_value(map, filter_values) do
-    cond do
-      is_nested_map?(map) ->
```

Comment:
I'm curious why you care if the map is nested; You're iterating over it and rebuilding it anyways. It's better just to fix things up while iterating.